akka-stream-testkit/src/main/scala/akka/stream/testkit/javadsl/TestKit.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.testkit.javadsl
+
+import akka.stream.Materializer
+import akka.stream.impl.PhasedFusingActorMaterializer
+import akka.stream.testkit.scaladsl
+
+object TestKit {
+
+  /**
+   * Stop all stages created by a given materializer.
+   */
+  def stopAllStages(mat: Materializer): Unit =
+    mat match {
+      case impl: PhasedFusingActorMaterializer ⇒
+        scaladsl.TestKit.stopAllChildren(impl.system, impl.supervisor)
+      case _ ⇒
+    }
+
+  /**
+   * Assert that there are no stages running under a given materializer.
+   * Usually this assertion is run after a test-case to check that all of the
+   * stages have terminated successfully.
+   */
+  def assertNoStages(mat: Materializer): Unit =
```

Comment:
why different name? `assertAllStagesStopped` ?

Reply:
The javadsl is split into two method calls. First one stopps all of the stages (meant to be ran before the test, as a precaution) and the second one asserts there are no stages left. Both are quite different in semantics from the scaladsl one which takes the test code block as a named parameter and does all the things. Therefore the different names for the javadsl.

Reply:
Without stopAllStages I think assertAllStagesStopped is appropriate naming for javadsl even though it doesn’t take a block.